bokeh/server/contexts.py

```diff
@@ -212,7 +213,12 @@ def create_session_if_needed(self, session_id, request=None):
             except Exception as e:
                 log.error("Failed to run session creation hooks %r", e, exc_info=True)
 
-            self._application.initialize_document(doc)
+            # using ThreadPoolExecutor create a thread that initializes new session document
+            try:
+                doc_initializer = concurrent.futures.ThreadPoolExecutor(max_workers=1)
+                doc_initializer.submit(self._application.initialize_document, doc)
+            except concurrent.futures.BrokenExecutor as e:
+                log.error("ThreadExecutor failed to create document for session with ID", session_id, e)
```

Comment:
Few issues with the code:
- The executor could be reused and initialized with a larger pool. Otherwise, you could just use `Thread` instead
- Now any exceptions thrown from `initialize_document` are ignored and not even logged (I think) since the future is resolved nowhere
- The immediate use of `ServerSession` below modifies the document without any logs - it can lead to some hard to catch concurrency issues
- `ServerSession` assumes that the document has already been initialized - it processes the document's callbacks

Maybe there are more potential issues somewhere down the line, there the result of `create_session_if_needed` is processed.